modules/core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/TestColdStorage.java

```diff
@@ -244,6 +244,43 @@ public void shouldCheckAvailability() {
         checkAvailabilityOfDocuments(Arrays.asList(documents.get(1), documents.get(2)), downloadableUntil, 0);
     }
 
+    @Test
+    public void shouldFailUpdateMainContentAlreadyInColdStorage() throws IOException {
+        // move the main content into the cold storage
+        DocumentModel documentModel = createFileDocument(DEFAULT_DOC_NAME, true);
+        moveAndVerifyContent(session, documentModel);
+
+        // we cannot update the main content as it is already in cold storage
+        documentModel = session.getDocument(documentModel.getRef());
+        documentModel.setPropertyValue(ColdStorageHelper.FILE_CONTENT_PROPERTY,
+                (Serializable) Blobs.createBlob(FILE_CONTENT));
+        try {
+            session.saveDocument(documentModel);
+            fail("Should fail because a main content document in cold storage cannot be updated.");
+        } catch (NuxeoException e) {
+            assertEquals(SC_FORBIDDEN, e.getStatusCode());
+            assertEquals(
+                    String.format("The main content of document: %s cannot be updated. It's already in cold storage.",
+                            documentModel.getId()),
+                    e.getMessage());
+        }
+
+        // but we should be able to update the others properties even the attachments
+        documentModel = session.getDocument(documentModel.getRef());
```

Comment:
not necessary I believe

Reply:
the reload of the document is needed otherwise my test fail, because before this part of the test I tried to update the blob, if i don't reload it i will find myself with a document contains the new blob.

Reply:
but we can use `refresh` instead of the `getDocument`